### DISCUSSION TITLE
partialConfig is missing the files array

### DISCUSSION THREAD

**Discussion by Wendenburg** [babel/babel#13402] [Q&A] [answered]
Hi all,

working in a complex monorepo with mixed structures, using js and typescript alongside. So it is hard to provide the full setup. In short: my repo was using Gatsby 2.x and Webpack 4.x, now I try to update to Gatsby 3.x and Webpack 5.x. My Build is working until the pages should be rendered, then it ends up with a poor message:

Cannot read property 'forEach' of undefined (in a cache file that does not even contain a loop.).

I debugged the webpack-error-utils in the Gatsby dist folder, that pointed out the babel-loader, that is also located in a Gatsby dist folder. Here I found out that **partialConfig** is an object, containing some data, but the **files** array is completely missing and causing the undefined error.

My question now: could you give me any hint under what conditions files are possibly undefined? What files should that array contain?
Any hint is much appreciated!

Thanks, Jo

**Comment by nicolo-ribaudo** [comment, marked as answer]
This is the PR that added `files` to the `partialConfig` objects: https://github.com/babel/babel/pull/11907. It is a `Set` containing all the config files loaded when to compile the file you are passing to Babel.

It was introduced in `@babel/core` 7.12.0: maybe you are using an older version?

**Reply by Wendenburg** [reply]
Thanks a lot @nicolo-ribaudo,  the root package.json still uses 7.11. Will try this out immediately.
